test/integration/IntegrationChecks.t.sol

```diff
@@ -0,0 +1,418 @@
+        assert_Snap_Added_OperatorWeight(operator, quorums,
+            "failed to add operator weight to operator and total stake in each quorum");
+
+        // IndexRegistry
+        assert_Snap_Added_OperatorCount(quorums,
+            "total operator count should have increased for each quorum");
+        assert_Snap_Added_OperatorListEntry(operator, quorums,
+            "operator list should have one more entry");
+
+        // DelegationManager
+        assert_IsRegisteredToAVS(operator,
+            "operator should be registered to AVS");
+    }
+
+    /// @dev Combines many checks of check_Register_State and check_Deregister_State
+    /// to account for quorums registered with churn
+    function check_Churned_State(
+        User incomingOperator,
+        User[] memory churnedOperators,
+        bytes memory churnedQuorums,
+        bytes memory standardQuorums
+    ) internal {
+        _log("check_Churned_State", incomingOperator);
+
+        bytes memory combinedQuorums = 
+            churnedQuorums
+                .orderedBytesArrayToBitmap()
+                .plus(standardQuorums.orderedBytesArrayToBitmap())
+                .bitmapToBytesArray();
+
+        // RegistryCoordinator
+        assert_HasOperatorInfoWithId(incomingOperator, 
+            "operatorInfo should have operatorId");
+        assert_HasRegisteredStatus(incomingOperator,
+            "operatorInfo status should be REGISTERED");
+        assert_IsRegisteredForQuorums(incomingOperator, combinedQuorums,
+            "current operator bitmap should include quorums");
+        assert_Snap_Registered_ForQuorums(incomingOperator, combinedQuorums,
+            "operator did not register for all quorums");
+
+        // BLSApkRegistry
+        assert_HasRegisteredPubkey(incomingOperator, 
+            "operator should have registered a pubkey");
+        assert_Snap_Added_QuorumApk(incomingOperator, standardQuorums,
+            "operator pubkey should have been added to standardQuorums apks");
+        assert_Snap_Churned_QuorumApk(incomingOperator, churnedOperators, churnedQuorums,
+            "operator pubkey should have been added and churned operator pubkeys should have been removed from apks");
+        
+        // StakeRegistry
+        assert_HasAtLeastMinimumStake(incomingOperator, combinedQuorums,
+            "operator should have at least the minimum stake in each quorum");
+        assert_Snap_Added_OperatorWeight(incomingOperator, standardQuorums,
+            "failed to add operator weight to operator and total stake in standardQuorums");
+        assert_Snap_Churned_OperatorWeight(incomingOperator, churnedOperators, churnedQuorums,
+            "failed to add operator weight and remove churned weight from each quorum");
+
+        // IndexRegistry
+        assert_Snap_Added_OperatorCount(standardQuorums,
+            "total operator count should have increased for standardQuorums");
```

Comment:
noting that this _could_ break in some circumstances, since standard quorums will have operators deregister to make room.
doesn't look to be breaking now, so probably a moot point though.

Reply:
This is true, but they were deregistered before the last snapshot was taken.

The snap only checks the prior state right before the user called registerOperatorWithChurn

Reply:
I was about to note that my understanding of the snapshot state was wrong when I made this comment, glad to see you corrected me already.
resolving now as I agree that this is good as-is!